metrics.go

```diff
@@ -0,0 +1,207 @@
+package metrics
+
+import (
+	"encoding/json"
+	"os"
+
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/alexcesaro/statsd"
+	"github.com/transcovo/go-chpr-logger"
+)
+
+var defaultClientBuilder *clientBuilder
+
+func defaultClientErrorHandler(err error) {
+	// Do we choose to make the metrics depend on the logger ?
+	// That sounds like a bad idea to me but this would be useful to have logs about the connection to the statsd server
+	// fmt.Println("Error caught in metrics", error)
+	logger.WithField("error", err).Error("[Metrics] Error caught in metrics")
+}
+
+func init() {
+	defaultClientBuilder = &clientBuilder{
+		errorLogger: defaultClientErrorHandler,
+	}
+}
+
+/*
+Struct responsible for keeping all the statsd clients
+*/
+type metricsSender struct {
+	Clients []*statsd.Client
+}
+
+var metrics *metricsSender = &metricsSender{}
```

Comment:
Didn't really dive in the code yet, but I find it hard to navigate in it with all the structs and the methods that are not declared next to each other.
I think I would prefer to split this file so that it contains only the entry functions (init, increment,...), and then have one file per struct and the methods, to easily navigate and find everything